mlir/utils/jenkins/MIOpenDriver.py

```diff
@@ -302,45 +303,75 @@ def benchmarkMIOpen(commandLine, xdlops):
     df.rename(columns={'TFlops': 'MLIR TFlops', 'TFlops (MIOpen)': 'MIOpen TFlops'}, inplace=True)
 
     df['MLIR/MIOpen'] = df['MLIR TFlops'] / df['MIOpen TFlops']
     df.to_csv(reportUtils.PERF_REPORT_FILE, index=False)
-
-    COLUMNS_TO_AVERAGE = ['MLIR TFlops', 'MIOpen TFlops', 'MLIR/MIOpen']
-    plotMean = df[COLUMNS_TO_AVERAGE].agg(reportUtils.geoMean)
-    plotMean.name = "Geo. mean"
-    plotMean = pd.DataFrame(plotMean).T
-
-    plotMean[['MLIR TFlops', 'MIOpen TFlops']]\
-        .to_csv(reportUtils.PERF_PLOT_REPORT_FILE, index=False)
-
-    means = df.groupby(["Direction", "DataType"])[COLUMNS_TO_AVERAGE]\
-        .agg(reportUtils.geoMean)
-    means.loc[("All", "All"),:] = df[COLUMNS_TO_AVERAGE].agg(reportUtils.geoMean)
-    means.to_csv(reportUtils.PERF_STATS_REPORT_FILE)
-
-    with open("MLIR_vs_MIOpen.html", 'w') as htmlOutput:
-        reportUtils.htmlReport(df, means, "MLIR vs. MIOpen performance", "MLIR/MIOpen", htmlOutput)
-
-def generatePerformanceResults(configs, xdlops):
-    mlir_df = pd.DataFrame(benchmarkMLIR(testVector.split(sep=' '), xdlops) for testVector in configs)
-    miopen_df = pd.DataFrame(benchmarkMIOpen(testVector.split(sep=' '), xdlops) for testVector in configs)
-
-    printPerformance(mlir_df, miopen_df)
+    with pd.option_context('precision', reportUtils.ROUND_DIGITS):
+        print(df)
+
+def getSolverName(testVector, xdlops):
+    config = ConvConfiguration.fromCommandLine(testVector.split(sep=' '), xdlops)
+    if config.direction == 'fwd':
+       solverName = 'ConvMlirIgemmFwd'
+    elif config.direction == 'bwd':
+       solverName = 'ConvMlirIgemmBwd'
+    else:
+       solverName = 'ConvMlirIgemmWrW'
+    if xdlops == True:
+       solverName+='Xdlops'
+    return solverName
+    
+def benchmarkMIOpenWithMLIRKernels(configs, xdlops):
+    solver_names = {testVector : getSolverName(testVector, xdlops) for testVector in configs}
+
+    # Set environment variables for running MIOpenDriver with MLIR kernels
+    envs = dict(MIOPEN_FIND_MODE='1', MIOPEN_DRIVER_USE_GPU_REFERENCE='l')
+    perf_list = []
+    for testVector in configs:
+        envs['MIOPEN_DEBUG_FIND_ONLY_SOLVER']=solver_names[testVector]
+        perf_list.append(benchmarkMIOpen(testVector.split(sep=' '), xdlops, envs))
+    df = pd.DataFrame(perf_list)
+    df.to_csv(reportUtils.MIOPEN_REPORT_FILE, index=False)
+
+#Tune MIOpen with MLIR kernels
+def tuneMLIRKernels(configs, xdlops):
+    solver_names = {testVector : getSolverName(testVector, xdlops) for testVector in configs}
+
+    envs = dict(MIOPEN_FIND_ENFORCE='4', MIOPEN_DRIVER_USE_GPU_REFERENCE='1')
+    for testVector in configs:
+        envs['MIOPEN_DEBUG_FIND_ONLY_SOLVER']=solver_names[testVector]
+        commandLine = testVector.split(sep=' ')
+        config = ConvConfiguration.fromCommandLine(commandLine, xdlops)
+        if config.inputLayout == 'nchw':
+            MIOpenDriverCommand = MIOPEN_DRIVER + ' ' + ' '.join(commandLine) + ' -V 0'
```

Comment:
Instead of string append and then a split(), use a list of arguments. That is, `[MIOPEN_DRIVER, *commandLine, '-V', '0']` or something along those lines.

Reply:
Good point. Thanks.